### DISCUSSION TITLE
Using Nuget without checking in nuget.exe

### DISCUSSION THREAD

**Discussion by mandalorianbob** [cake-build/cake#3399] [Q&A] [answered]
I've got a Cake.Frosting build runner and a private nuget feed.

I'd like to use Nuget to restore via the private feed, but Cake's unable to find nuget.exe. This is probably because I don't really have any local copies. I'd rather not store nuget.exe is my source, but I'm not sure how else to reliably get and use nuget in the context of a Frosting runner.

Am I missing an option somewhere?

**Comment by augustoproiete** [comment, marked as answer]
@mandalorianbob An easy way to make `nuget.exe` available to Cake is to install the package [`NuGet.CommandLine`](https://www.nuget.org/packages/NuGet.CommandLine/) as a tool via [`InstallTool`](https://cakebuild.net/docs/writing-builds/tools/installing-tools).

```csharp
public static class Program
{
    public static int Main(string[] args)
    {
        return new CakeHost()
            .InstallTool(new Uri("nuget:?package=NuGet.CommandLine&version=5.9.1"))
            .Run(args);
    }
}
```

**Reply by mandalorianbob** [reply]
This works. I didn't think of the nuget.exe being separate from the nuget that you use to download packages, but that makes sense.

Thanks!
